api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -123,45 +129,22 @@ public Patient registerPatient(Patient patient, List<Relationship> relationships
 			log.info("Registering new patient..");
 		if (patient == null)
 			throw new APIException("Patient cannot be null");
-		
-		IdentifierSourceService iss = Context.getService(IdentifierSourceService.class);
-		if (idSource == null) {
-			String idSourceId = adminService.getGlobalProperty(RegistrationCoreConstants.GP_IDENTIFIER_SOURCE_ID);
-			if (StringUtils.isBlank(idSourceId))
-				throw new APIException("Please set the id of the identifier source to use to generate patient identifiers");
-			
-			try {
-				idSource = iss.getIdentifierSource(Integer.valueOf(idSourceId));
-				if (idSource == null)
-					throw new APIException("cannot find identifier source with id:" + idSourceId);
-			}
-			catch (NumberFormatException e) {
-				throw new APIException("Identifier source id should be a number");
-			}
-		}
-		
-		if (identifierLocation == null) {
-			identifierLocation = locationService.getDefaultLocation();
-			if (identifierLocation == null)
-				throw new APIException("Failed to resolve location to associate to patient identifiers");
-		}
 
-        // generate identifier if necessary, otherwise validate
-        if (StringUtils.isBlank(identifierString)) {
-            identifierString = iss.generateIdentifier(idSource, null);
-        }
-        else {
-            PatientIdentifierValidator.validateIdentifier(identifierString, idSource.getIdentifierType());
-        }
-		PatientIdentifier pId = new PatientIdentifier(identifierString, idSource.getIdentifierType(), identifierLocation);
-		patient.addIdentifier(pId);
-        pId.setPreferred(true);
+		Integer openMrsIdentifierId = coreProperties.getIdentifierSourceId();
+		PatientIdentifier patientIdentifier;
+		if (StringUtils.isBlank(identifierString)) {
+			patientIdentifier = identifierBuilder.generateIdentifier(openMrsIdentifierId, identifierLocation);
+		} else {
+			patientIdentifier = identifierBuilder.createIdentifier(openMrsIdentifierId, identifierString, identifierLocation);
```

Comment:
at first glance it seems like you've dropped the validation step, but perhaps it's done within this method.